app/models/pieces/pawn.rb

```diff
@@ -1,8 +1,36 @@
 class Pawn < Piece
   before_create :set_defaults
+  #attr_accessor :first_move
   
   def set_defaults
     self.icon = (self.color == 1) ? "&#9823;" : "&#9817;"
+    self.first_move ||= true
   end
-
+  
+  def valid_move?(x, y)
+    # Valid parameters passed in?
+    valid_params = super
+    return false if !valid_params
```

Comment:
Just a gimmick of ruby - you can use unless which means 'if not' - so the line would be:

return false unless valid_params